datarepo-clienttests/src/main/java/scripts/testscripts/RetrieveSnapshot.java

```diff
@@ -34,14 +41,54 @@ public RetrieveSnapshot() {
 
   private SnapshotSummaryModel snapshotSummaryModel;
   private List<BlobId> scratchFiles = new ArrayList<>();
+  private boolean deleteProfile;
 
   public void setup(List<TestUserSpecification> testUsers) throws Exception {
-    // create the profile and dataset
-    super.setup(testUsers);
+    // Specify the name of a profile to use for the test with the TEST_RUNNER_BILLING_PROFILE_NAME
+    // env variable.
+    // If it's not specified or the profile isn't found, a new one with a randomly generated name
+    // will be created
+    String profileNameRaw = System.getenv("TEST_RUNNER_BILLING_PROFILE_NAME");
+    Optional<String> profileName;
+    if (StringUtils.isBlank(profileNameRaw)) {
+      profileName = Optional.empty();
+      deleteProfile = true;
+      logger.info("No profile name specified.  Will create a new billing profile");
+    } else {
+      profileName = Optional.ofNullable(profileNameRaw);
+      deleteProfile = false;
+      logger.info("Will attempt to use a billing profile named {}", profileName.get());
+    }
+
+    // pick the a user that is a Data Repo steward to be the dataset creator
+    datasetCreator = SAMUtils.findTestUserThatIsDataRepoSteward(testUsers, server);
 
     // get the ApiClient for the snapshot creator, same as the dataset creator
     ApiClient datasetCreatorClient = DataRepoUtils.getClientForTestUser(datasetCreator, server);
     RepositoryApi repositoryApi = new RepositoryApi(datasetCreatorClient);
+    ResourcesApi resourcesApi = new ResourcesApi(datasetCreatorClient);
+    BillingProfileModel billingProfileModel =
+        profileName
+            .map(
```

Comment:
I continue to be un-sold on using `Optional` in this way. using `profileName.map(p -> { ... })` to seems less clear than:
```
if (profileName != null) {
    ...
} else {
   ...
}
```

Reply:
ok fine :) One day we can do all the things in fluent format